lib/process/watch.go

```diff
@@ -35,10 +35,10 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-// startCertificateWatch starts watching for the changes in cluster certificate
-// and notifies the process' "certificateCh" when the change happens
-func (p *Process) startCertificateWatch(ctx context.Context, client *kubernetes.Clientset) error {
-	go func() {
+// runCertificateWatch updates process on p.certificateCh
+// when changes to cluster certificates are detected
+func (p *Process) runCertificateWatch(client *kubernetes.Clientset) clusterService {
```

Comment:
Is this now a "cluster service"? If so, it will run only in active gravity-site, but should run in all.

Some goroutines should only run in active gravity-site (e.g. status checker) and stop when it loses leadership, but some should always run in all gravity-sites (e.g. certificate/authgateway watches).

Reply:
These continue to function as expected - those that are only supposed to run on the leader, only do run on the leader (this requirement is part of the `RegisterClusterService` API which adds leader-specific services). The only change is that there's a bunch of similar looking "service" functions which I moved under `cluster service` umbrella which does not mean leader-specific service.